### ISSUE TITLE
Unable to use use the output of this module to write local files

### ISSUE THREAD

**Issue by vikaskoppineedi** [hashicorp/terraform-template-dir#2]
With the Below Terraform Block , I get the following error. Can anyone suggest? 

Terraform v0.12.28

![image](https://user-images.githubusercontent.com/5601825/87895748-d67a1f00-ca0b-11ea-964e-9bb95fca715a.png)

![image](https://user-images.githubusercontent.com/5601825/87896027-a7b07880-ca0c-11ea-9dbb-5000b36086eb.png)

**Comment by vikaskoppineedi** [comment]
When I comment the aws_iam_role.eks_worker_role.arn and give "blah-blah", it works. How do i create depends on for module. 
![image](https://user-images.githubusercontent.com/5601825/87903942-dafe0200-ca22-11ea-8ed3-d5e78bf2a5ac.png)

**Comment by Myafq** [comment]
@vikaskoppineedi 
It's something about the declarative nature of Terraform. I also stumbled upon this issue and solved it with a two-steps terraform run approach. 
Usually, cloud resources are created only during very first terraform apply, and then we just making some changes to whatever files we have rendered. 

```
module "template_example" {
  source = "apparentlymart/dir/template"

  base_dir             = "${path.module}/k8s-system-example"
  template_file_suffix = ".yaml"
  template_vars = {
    env_folder = local.env_folder
  }
}

resource "local_file" "k8s-example" {
  for_each             = var.render_manifests ? module.template_example.files : {}
  content              = each.value.content
  file_permission      = "0644"
  directory_permission = "0755"
  filename             = "${path.cwd}/k8s-system-example/${each.key}.yaml"
}

```
We're using `var.render_manifest` to switch between two modes - cloud resource generation and manifests rendering.
First run - render=off. After cloud resources are ready - render=on.

**Comment by apparentlymart** [comment]
Hi @vikaskoppineedi,

I think what you are seeing here is a bug in Terraform itself: hashicorp/terraform#24024. That bug was fixed as part of the v0.13.0 development (specifically, as part of implementing module `for_each`) so a fix will be available in a couple of weeks when v0.13.0 final is released.

(v0.13.0-rc1 is currently available if you'd like to test it, but I would suggest only running `terraform plan` with that version, because running `terraform apply` will upgrade your state to be incompatible with Terraform v0.12.28.)

Unfortunately there is another problem which prevents fully using this module with the `local_file` resource: this module is designed to be used with resources that support both file contents given as a string _and_ references to existing files on disk. [`aws_s3_bucket_object`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_object), for example, has both `source` and `content` arguments to support both result types from this module, but `local_file` only has `content`.

That limitation doesn't matter if your directory contains _only_ template files, because those will always be returned as content strings. It will fail if you have at least one non-template file in the directory, because those are always returned as a direct reference to the existing file on disk. However, you could use the other output value `files_in_memory` to get _only_ the subset of files that are compatible with `local_file` and thus avoid an error if there is a non-template file. (the non-template files will just be ignored entirely, then)

Unfortunately there's nothing we can change in _this_ repository to make the `local_file` resource compatible with it, so if you're interested in full support for `local_file` with this module you could open a feature request for it in [the `local` provider repository](https://github.com/hashicorp/terraform-provider-local).

Because the problem described in the original issue here is fixed by Terraform v0.13, and because full support for the `local_file` requires changes in the `local` provider rather than in this module, I'm going to close this. Thanks for reporting this!
